net/dnscache/dnscache_test.go

```diff
@@ -140,3 +140,26 @@ func TestResolverAllHostStaticResult(t *testing.T) {
 		t.Errorf("bad dial error got %q; want %q", got, want)
 	}
 }
+
+func TestInterleaveSlices(t *testing.T) {
+	testCases := []struct {
+		a, b []int
+		want []int
+	}{
+		{a: []int{1, 3, 5}, b: []int{2, 4, 6}, want: []int{1, 2, 3, 4, 5, 6}},
+		{a: []int{1, 3, 5}, b: []int{2, 4}, want: []int{1, 2, 3, 4, 5}},
+		{a: []int{1, 3}, b: []int{2, 4, 6}, want: []int{1, 2, 3, 4, 6}},
+		{a: []int{1}, b: []int{2, 4, 6}, want: []int{1, 2, 4, 6}},
+		{a: nil, b: []int{2, 4, 6}, want: []int{2, 4, 6}},
+		{a: nil, b: nil, want: nil},
+	}
+
+	for i, tc := range testCases {
+		t.Run(fmt.Sprintf("TestCase%d", i), func(t *testing.T) {
```

Comment:
test cases should be named, numbering is annoying because it requires counting when debugging.